src/Test.UnitTests.Sarif.Driver/HttpMockHelper.cs

```diff
@@ -0,0 +1,101 @@
+            new HttpResponseMessage(HttpStatusCode.OK);
+
+        public static readonly HttpResponseMessage NotFoundResponse =
+            new HttpResponseMessage(HttpStatusCode.NotFound);
+
+        public static readonly HttpResponseMessage ForbiddenResponse =
+            new HttpResponseMessage(HttpStatusCode.Forbidden);
+
+        public static readonly HttpResponseMessage BadRequestResponse =
+            new HttpResponseMessage(HttpStatusCode.BadRequest);
+
+        public static readonly HttpResponseMessage UnauthorizedResponse =
+            new HttpResponseMessage(HttpStatusCode.Unauthorized);
+
+        public static readonly HttpResponseMessage InternalServerErrorResponse =
+            new HttpResponseMessage(HttpStatusCode.InternalServerError);
+
+        public static readonly HttpResponseMessage NonAuthoritativeInformationResponse =
+            new HttpResponseMessage(HttpStatusCode.NonAuthoritativeInformation);
+
+        private readonly List<Tuple<HttpRequestMessage, HttpResponseMessage>> _fakeResponses =
+            new List<Tuple<HttpRequestMessage, HttpResponseMessage>>();
+
+        public void Mock(HttpRequestMessage httpRequestMessage, HttpStatusCode httpStatusCode, HttpContent httpContent)
+        {
+            _fakeResponses.Add(new Tuple<HttpRequestMessage, HttpResponseMessage>(
+                httpRequestMessage,
+                new HttpResponseMessage(httpStatusCode) { RequestMessage = httpRequestMessage, Content = httpContent }));
+        }
+
+        public void Mock(HttpRequestMessage httpRequestMessage, HttpResponseMessage httpResponseMessage)
+        {
+            _fakeResponses.Add(new Tuple<HttpRequestMessage, HttpResponseMessage>(
+                httpRequestMessage,
+                httpResponseMessage));
+        }
+
+        public void Clear()
+        {
+            _fakeResponses.Clear();
+        }
+
+        private static bool CompareHeaders(HttpRequestHeaders headers1, HttpRequestHeaders headers2)
+        {
+            foreach (KeyValuePair<string, IEnumerable<string>> header in headers1)
+            {
+                string headerName = header.Key;
+                if (!headers2.TryGetValues(headerName, out IEnumerable<string> values))
+                {
+                    return false;
+                }
+
+                string headerContent1 = string.Join(",", header.Value);
+                string headerContent2 = string.Join(",", values);
+
+                if (headerContent1 != headerContent2)
+                {
+                    return false;
+                }
+            }
+
+            return true;
+        }
+
+        protected override Task<HttpResponseMessage> SendAsync(HttpRequestMessage request, CancellationToken cancellationToken)
+        {
+            Tuple<HttpRequestMessage, HttpResponseMessage> fakeResponse;
+
+            if (request.Headers.IsEmptyEnumerable())
+            {
+                fakeResponse = _fakeResponses.Find(fr =>
```

Comment:
>_fakeResponses [](http://example.com/codeflow?start=31&length=14)

our code is a mixture of this. and underscore prefixed patterns./ we need to stop this inconsistency and land on a standard.

Reply:
let's use this. :)